### PR TITLE
add bash completion

### PR DESCRIPTION
**Pull request by jrwren**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/104/
Test PASSed.

File: Makefile [hunk 1]
```diff
@@ -82,23 +82,38 @@ man/man1:
...
 	 	rm "$(DESTDIR)/usr/share/$$file" ; done
+	-rmdir -p $(DESTDIR)/usr/share/man/man1
+
+install-bash-completion:
+	mkdir -p $(DESTDIR)/usr/share/bash-completion/completions
+	$(INSTALL_FILE) config/bash/charm "$(DESTDIR)/usr/share/bash-completion/completions/"
+
+uninstall-bash-completion:
+	rm "$(DESTDIR)/usr/share/bash-completion/completions/charm"
+	-rmdir -p $(DESTDIR)/usr/share/bash-completion/completions
```
Comment (bac) [inline Makefile:100]: Whoa there!

Why do you want to remove all completions?  My system has 780 in that directory.  Even we created the directory we can't be sure there isn't other stuff in there and just leaving the directory doesn't hurt anything except our sense of hygiene.

File: Makefile [hunk 2]
```diff
@@ -82,23 +82,38 @@ man/man1:
...
 	@echo 'make clean - Remove object files from package source directories.'
 	@echo 'make deps - Set up the project Go dependencies.'
 	@echo 'make create-deps - Generate the Go dependencies file.'
 	@echo 'make format - Format the source files.'
 	@echo 'make man - Generate man pages.'
 	@echo 'make simplify - Format and simplify the source files.'
+	@echo 'make install-man - Install man pages to $DESTDIR/usr/share/man'
+	@echo 'make uninstall-man - Remove man pages from $DESTDIR/usr/share/man'
+	@echo 'make install-bash-completion - Install completion to $DESTDIR//usr/share/bash-completion/completions/'
+	@echo 'make uninstall-bash-completion - Remove completion from $DESTDIR//usr/share/bash-completion/completions/'
```
Comment (bac) [inline Makefile]: Double // in the previous two entries

Comment (bac) [inline Makefile]: The use of variables in these help commands is not working.  I see

make install - Install the package to OPATH/bin
make install-man - Install man pages to ESTDIR/usr/share/man

**Comment by bac** [comment]
QA OK:
charm <tab> shows subcommands
charm publish <tab> shows options.

Thanks!

File: Makefile (hunk 1 above)
Comment (jrwren) [inline Makefile:100]: Its rmdir -p, not rm -rf. This is standard behavior I shoplifted from autotools generated makefiles.

rmdir only removes empty directories, so in the off-chance that we created this directory, or the entire heirarchy on install-bash-completion, we remove it in uninstall-bash-completion.

File: Makefile (hunk 2 above)
Comment (jrwren) [inline Makefile]: Fixes. TY

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/105/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/106/
Test PASSed.

**Comment by mhilton** [comment]
:+1: I think, I'm not totally sure I understood it all.

**Comment by jrwren** [comment]
:shipit:

**Comment by jrwren** [comment]
I don't admit to understanding it all either. I reduced the juju completion script to the smallest thing that made sense to me.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**